lib/cpu_gauge_field.cpp

```diff
@@ -173,7 +173,7 @@ namespace quda {
     if (link_direction != QUDA_LINK_BACKWARDS)
       errorQuda("link_direction = %d not supported", link_direction);
 
-    void *recv[QUDA_MAX_DIM];
+    void *recv[2*QUDA_MAX_DIM];
```

Comment:
just by looking at the use of recv: What motivated the change to 2*QUDA_MAX_DIM. This is not obvious to me.

Reply:
This comes from the constructor for the QDP fine-grain accessor (gauge_field_order.h).  It assumes the ghost array is of length `2*nDim`.  The upper elements are never used, but this silences the ASAN error.